private/buf/cmd/buf/command/beta/registry/draft/draftlist/draftlist.go

```diff
@@ -106,27 +107,124 @@ func run(
 	if err != nil {
 		return err
 	}
-	service := connectclient.Make(
-		clientConfig,
-		moduleFullName.Registry(),
-		registryv1alpha1connect.NewRepositoryCommitServiceClient,
+	moduleServiceClient := bufapi.NewClientProvider(clientConfig).V1ModuleServiceClient(moduleFullName.Registry())
+	moduleResp, err := moduleServiceClient.GetModules(
+		ctx,
+		&connect.Request[modulev1.GetModulesRequest]{
+			Msg: &modulev1.GetModulesRequest{
+				ModuleRefs: []*modulev1.ModuleRef{
+					{
+						Value: &modulev1.ModuleRef_Name_{
+							Name: &modulev1.ModuleRef_Name{
+								Owner:  moduleFullName.Owner(),
+								Module: moduleFullName.Name(),
+							},
+						},
+					},
+				},
+			},
+		},
 	)
-	resp, err := service.ListRepositoryDraftCommits(
+	if err != nil {
+		if connect.CodeOf(err) == connect.CodeNotFound {
+			return bufcli.NewRepositoryNotFoundError(container.Arg(0))
+		}
+		return err
+	}
+	modules := moduleResp.Msg.GetModules()
+	if len(modules) != 1 {
+		return syserror.Newf("expected 1 module from response, got %d", len(modules))
+	}
+	defaultLabelName := modules[0].GetDefaultLabelName()
+	labelServiceClient := bufapi.NewClientProvider(clientConfig).V1LabelServiceClient(moduleFullName.Registry())
+	order := modulev1.ListLabelsRequest_ORDER_CREATE_TIME_ASC
+	if flags.Reverse {
+		order = modulev1.ListLabelsRequest_ORDER_CREATE_TIME_DESC
+	}
+	labelsResp, err := labelServiceClient.ListLabels(
 		ctx,
-		connect.NewRequest(&registryv1alpha1.ListRepositoryDraftCommitsRequest{
-			RepositoryOwner: moduleFullName.Owner(),
-			RepositoryName:  moduleFullName.Name(),
-			PageSize:        flags.PageSize,
-			PageToken:       flags.PageToken,
-			Reverse:         flags.Reverse,
-		}),
+		&connect.Request[modulev1.ListLabelsRequest]{
+			Msg: &modulev1.ListLabelsRequest{
```

Comment:
It feels like there's a bug in the API, can someone explain to me what is going on here? @saquibmian @unmultimedio

Reply:
`ListLabels` will return all labels that exist. This includes the default label, if it exists (when the repository is first created, no labels exist, not even the default label).

The page token logic is a little burdensome here... In alpha RPCs for back-compat we specifically exclude the default label. Maybe we need a field in the request that allows callers to skip the default label? `bool exclude_default_label`, with a default `false`?

Reply:
> In alpha RPCs for back-compat we specifically exclude the default label

Can you explain this a bit more?

Reply:
`ListRepositoryDraftCommitsRequest` used to return all drafts, which of course never included "main"/the default label.
Once drafts were migrated to Labels, `ListRepositoryDraftCommitsRequest` now returns all labels _except_ the default label, because the default label should never be a draft. That's the behavior that Oliver is intending to maintain here as well.

Reply:
OK - @oliversun9 I know I told you to do this in two pieces, but can you instead just make a new `buf beta registry label` subcommand and do the expected thing for list/get/etc for labels specifically, and delete `buf beta registry draft/tag`? I don't want us spending cycles on getting the old beta commands for drafts/tags being backwards-compatible when we're just going to delete them anyways.

Reply:
Sure, that sounds good